db/migrate/20230503192155_sync_log_has_unique_state.rb

```diff
@@ -0,0 +1,14 @@
+class SyncLogHasUniqueState < ActiveRecord::Migration[7.0]
+  def change
+    reversible do |dir|
+      dir.up do
+        SyncLog.where.not(id: SyncLog.latest).delete_all
+      end
+    end
+
+    add_index :sync_logs, [:feeder_type, :feeder_id], unique: true
```

Comment:
Ah, I see.  So there's guaranteed to only be one of these.

Noting that these SyncLogs aren't namespaced to "apple", but are generally treated as if they're unique to apple.  If we want to expand them to handle other external systems, will need to add some other column/scope.  (But that's fine - one thing at a time!)